PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/hdinsight/spark/ui/SparkSubmissionContentPanel.kt

```diff
@@ -567,6 +600,46 @@ open class SparkSubmissionContentPanel(private val myProject: Project, val type:
 
     override fun dispose() {
     }
+
+    fun prepareFileSystem(uploadRootPath: String?): AzureStorageVirtualFile? {
+        referencedJarsTextField.setButtonEnabled(false)
+        fileSystem = null
+        //val cluster = clustersSelection.viewModel.clusterComboBoxSelection as? ClusterDetail
+
+        val cluster = clustersSelection.viewModel.clusterIsSelected
+                .toBlocking()
+                .firstOrDefault(null)
+
+        var storageAccount: IHDIStorageAccount? = cluster?.storageAccount
+        try {
+            when (storageWithUploadPathPanel.storagePanel.viewModel.deployStorageTypeSelection) {
+                SparkSubmitStorageType.DEFAULT_STORAGE_ACCOUNT -> fileSystem = if (storageAccount?.accountType == StorageAccountType.ADLSGen2)
+                    ADLSGen2FileSystem(SharedKeyHttpObservable(storageAccount.name,
+                            (storageAccount as? ADLSGen2StorageAccount)?.primaryKey),
+                            uploadRootPath) else null
+
+                SparkSubmitStorageType.ADLS_GEN2 -> {
+                    val host = URI.create(uploadRootPath).host
+                    val account = host.substring(0, host.indexOf("."))
+                    var accessKey = storageWithUploadPathPanel.storagePanel.adlsGen2Card.storageKeyField.text.trim()
+
+                    if (StringUtils.isBlank(host) || StringUtils.isBlank(accessKey)) {
+                        return null
+                    }
+
+                    fileSystem = ADLSGen2FileSystem(SharedKeyHttpObservable(account, accessKey), uploadRootPath)
+                }
+                else -> {
+                }
+            }
+        } catch (ex: IllegalArgumentException) {
+            log().warn("Preparing file system encounter ", ex)
+        }
+
+        val gen2System = fileSystem as? ADLSGen2FileSystem
+        return if (gen2System != null) AdlsGen2VirtualFile(gen2System.root, true, gen2System)
```

Comment:
The code of 2 lines above can be moved into `when()  { SparkSubmitStorageType.ADLS_GEN2 -> { ...` block